### PR TITLE
Add Detection of Voice with RNNoise

### PR DESCRIPTION
**Pull request by diegocardozo97**
Add algorithm to detect when a voice starts and ends in the input audio, clear it of noice, and publish the cleaned piece of audio with voice. All this by calling _RNNoise_.
- Algorithm to detect voice in _audio_
- Clear noise
- Publish voice to _UsefulAudio_

**Note**: The actual code of RNNoise is not yet in the repo. Also, work needs to be done to ensure full compatibility with the format of ROS' _audiodata_.

InputAudio-Node Diagram

### CODE REVIEW

File: catkin_home/src/devices/src/InputAudio.cpp [hunk 1]
```diff
@@ -1,26 +1,326 @@
...
+    output_vector.insert(output_vector.end(),
+      (uint8_t*)(recording + begin_index_element),
+      (uint8_t*)(recording + RECORDING_BUFFER_SIZE));
+    output_vector.insert(output_vector.end(), 
+      (uint8_t*)recording,
+      (uint8_t*)(recording + end_index_element));
+  }
+
+  audio_common_msgs::AudioData msg_audio;
+  msg_audio.data = output_vector;
```
Comment (paulvazbad) [inline catkin_home/src/devices/src/InputAudio.cpp]: No da problemas de formato con ROS ? Que podemos hacer para atacar la nota en tu commit de la compatibilidad?

Comment (paulvazbad) [reply catkin_home/src/devices/src/InputAudio.cpp]: Do you have information regarding the setup process of RNNoise ? I believe including it in the repo is necessary.

**Comment by diegocardozo97** [comment]
I already edited my initial comment with a simple diagram that describes the input and output of the InputAudio node.

File: catkin_home/src/devices/src/InputAudio.cpp (hunk 1 above)
Comment (diegocardozo97) [reply catkin_home/src/devices/src/InputAudio.cpp]: > No da problemas de formato con ROS ? Que podemos hacer para atacar la nota en tu commit de la compatibilidad?

I actually don't know. I am going to do the complete integration in a following pull request.



> Do you have information regarding the setup process of RNNoise ? I believe including it in the repo is necessary.

It is actually "just" a library. Anyway, I am going to include about it in my next pull request.

**Pull request merged**